app/controllers/api/v1/search_controller.rb

```diff
@@ -619,6 +632,29 @@ def self.match_studies_by_facet(query_results, search_facets)
         matches
       end
 
+      # Simplify TDR results to be mappable for the UI badges for facet search
+      def self.simplify_TDR_facet_search_results(query_results, search_facets)
+        simple_TDR_result = {}
+        simple_TDR_results = []
+        query_results.each do |result|
```

Comment:
Since `query_results` is a Hash, we can iterate over the keys & values natively like below.  In this case, the keys are "accessions", or project short names.  This way you don't need to assign `results_details` as it will automatically be `result`:
```suggestion
        query_results.each do |accession, result|
```